contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+or CSI operators. Hiding the container mountpoints reduces the steady-state
+systemd CPU utilization from ~10% to ~0% on every worker node. On a more
+heavily loaded system, with 1000’s of container mountpoints, systemd’s usage
+will cap out at 100% of a single CPU core. Even with the 30-50% reduction in
+systemd usage expected from the December 2020 fix, having systemd take up as
+much as 50-70% of a single CPU in the worst-case scenario is a lot of
+unnecessary overhead.
+
+Administrators of other OSes with other host OS services that monitor or
+process mountpoints indiscriminately could also benefit from hiding these
+container-specific mountpoints from the OS at large.
+
+### Tidiness and Encapsulation
+
+These container overlay mounts, secret and configmap mounts, and namespace
+mounts are all internal implementation details around the way the Kubernetes
+and the Container Runtime allocate resources to containers. For normal
+day-to-day operation, there is no need for these special-use mounts to be
+visible in the host OS's top-level 'mount' or 'findmnt' output.
+
+### Security
+
+While access to container-specific mounts in Kubernetes (like secrets) today
+are protected by filesystem permissions, the fact that the locations are
+visible to unprivileged users via the 'mount' command can provide information
+which could be used in conjunction with other vulnerabilities to decipher
+secrets or other protected container mount contents. Locking access to even
+listing the mountpoints behind the ability to enter the mount namespace (which
+in turn requires `CAP_SYS_CHROOT` and `CAP_SYS_ADMIN`) removes one level of
+potential access to container-specific mounts.
+
+### No one uses it (that we know of)
+
+After much investigation and conversation on both the sig-node and sig-storage
+mailing lists, as well as within Red Hat's OpenShift communities, we have yet
+to turn up a use case that relies on this container-to-host propagation.
+
+### No one _should_ use it
+
+In fact, it may be seen as an anti-pattern and run contrary to the
+container-native philosophy. Relying on a host OS to interact with a
+container-owned mountpoint seems suspect.
+
+### Testing implications
+
+With separate mount namespaces, it would be possible to run multiple instances
+of Kubernetes on a single host without the various mountpoints interfering with
+one another. There may be other inter-instance difficulties besides the mount
+namespace, but this does move in that direction without resorting to more
+complicated solutions like trying to run Kubernetes inside of containers.
+
+## Costs and Implications
+
+If a Kubernetes distribution intends to implement this mount-hiding mode of
+operation, there are a number of costs and implications that should be
+documented for system administrators.
+
+### Privileged pods
+
+Privileged pods that previously expected to have direct access to the host's
+root mount namespace (such as the one spawned by `oc debug`), will now be
```

Comment:
Do we have a kubectl specific use case here we could use?

Reply:
Not that I know of, but I'm open to suggestions.

Reply:
@SergeyKanzhelev is there an upstream example we could use here?